### PR TITLE
Fix encode MySQLHandshakePacket UT

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

Previous the UT depends on `KYUUBI_VERSION`, so bump version casue it to fail.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-server/src/test/scala/org/apache/kyuubi/server/mysql/authentication/MySQLAuthPacketSuite.scala [hunk 1]
```diff
@@ -31,15 +31,17 @@ class MySQLAuthPacketSuite extends KyuubiFunSuite with MySQLCodecHelper {
   }
 
   test("encode MySQLHandshakePacket") {
-    val packet = MySQLHandshakePacket(2, authPluginData)
+    val packet = new MySQLHandshakePacket(2, authPluginData) {
+      override def serverVersion: String = "5.7.22-Kyuubi-Server v1.4.0-incubating"
```
Comment (turboFei) [inline kyuubi-server/src/test/scala/org/apache/kyuubi/server/mysql/authentication/MySQLAuthPacketSuite.scala:35]: should we use `dummy` here?

```
      override def serverVersion: String = "dummy"
```

Comment (pan3793) [reply kyuubi-server/src/test/scala/org/apache/kyuubi/server/mysql/authentication/MySQLAuthPacketSuite.scala:35]: Make sense

**Review by turboFei — APPROVED** [review]

**Review by yaooqinn — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merged to master

**Comment by yaooqinn** [comment]
we need  in branch-1.4 as it will also have new versions

**Pull request closed without merging**